admin/includes/modules/products_previous_next.php

```diff
@@ -1,151 +1,151 @@
-      $prev_next_order
-  ;
-
-  $products_ids = $db->Execute($sql);
-}
-
-// reset if not already set for display
-(isset($_GET['products_filter']) && $_GET['products_filter'] == '' ? (int)$_GET['products_filter'] = $products_filter : '');
-(isset($_GET['current_category_id']) && $_GET['current_category_id'] == '' ? (int)$_GET['current_category_id'] = $current_category_id : '');
-
-$id_array = [];
-foreach ($products_ids as $products_id) {
-  $id_array[] = $products_id['products_id'];
-}
-
-$counter = 0;
-// if invalid product id skip
-$id_array_size = count($id_array);
-if ($id_array_size) {
-  foreach ($id_array as $key => $value) {
-    if ($value == $products_filter) {
-      $position = $counter;
-      if ($key == 0) {
-        $previous = -1; // it was the first to be found
-      } else {
-        $previous = $id_array[$key - 1];
-      }
-      if ((($key + 1) < $id_array_size) && $id_array[$key + 1]) {
-        $next_item = $id_array[$key + 1];
-      } else {
-        $next_item = $id_array[0];
-      }
-    }
-    $last = $value;
-    $counter++;
-  }
-
-  if ($previous == -1) {
-    $previous = $last;
-  }
-
-  $sql = "SELECT categories_name
-          FROM " . TABLE_CATEGORIES_DESCRIPTION . "
-          WHERE categories_id = " . (int)$current_category_id . "
-          AND language_id = " . (int)$_SESSION['languages_id'];
-
-  $category_name_row = $db->Execute($sql);
-} // if id_array
-
-/*
-  if (strstr($PHP_SELF, FILENAME_PRODUCTS_PRICE_MANAGER)) {
-  $curr_page = FILENAME_PRODUCTS_PRICE_MANAGER;
-  } else {
-  $curr_page = FILENAME_ATTRIBUTES_CONTROLLER;
-  }
- */
-
-switch (true) {
-  case (strstr($PHP_SELF, FILENAME_ATTRIBUTES_CONTROLLER)):
-    $curr_page = FILENAME_ATTRIBUTES_CONTROLLER;
-    break;
-  case (strstr($PHP_SELF, FILENAME_PRODUCTS_TO_CATEGORIES)):
-    $curr_page = FILENAME_PRODUCTS_TO_CATEGORIES;
-    break;
-  default:
-    $curr_page = FILENAME_PRODUCTS_PRICE_MANAGER;
-    break;
-}
-// to display use products_previous_next_display.php
+<?php
+
+/**
+ * @copyright Copyright 2003-2020 Zen Cart Development Team
+ * @copyright Portions Copyright 2003 osCommerce
+ * @license http://www.zen-cart.com/license/2_0.txt GNU Public License V2.0
+ * @version $Id: DrByte 2019 Oct 01 Modified in v1.5.7 $
+ */
+if (!defined('IS_ADMIN_FLAG')) {
+  die('Illegal Access');
+}
+
+/////
+// BOF PREVIOUS NEXT
+
+if (!isset($current_category_id)) {
+  $current_category_id = 0;
+}
+
+if (!isset($prev_next_list) || $prev_next_list == '') {
+// calculate the previous and next
+
+  $result = $db->Execute("SELECT products_type
+                          FROM " . TABLE_PRODUCTS . "
+                          WHERE products_id = " . (int)$products_filter);
+  $check_type = ($result->EOF) ? 0 : $result->fields['products_type'];
+  if (!defined('PRODUCT_INFO_PREVIOUS_NEXT_SORT')) define('PRODUCT_INFO_PREVIOUS_NEXT_SORT', zen_get_configuration_key_value_layout('PRODUCT_INFO_PREVIOUS_NEXT_SORT', $check_type));
```

Comment:
For anyone seeking to further incorporate available Zen Cart created functions into this code, per #5547, see below about doing using the function `zen_define_default` (available as of Zen Cart 1.5.8) for replacement of this line such as:
````php
zen_define_default('PRODUCT_INFO_PREVIOUS_NEXT_SORT', zen_get_configuration_key_value_layout('PRODUCT_INFO_PREVIOUS_NEXT_SORT', $check_type));
```` 
> The creation of a debug log in a case like this seems proper to me. Your database has been damaged (or your code is bad).

There is not yet a function available to test if the `configuration_key` exists in the table `product_type_layout`. So per the above, you have to either test for the existence of that `configuration_key` initially, blindly generate the key, or accept the debug log that will be created on each page load. Why one would do this testing up front and not just carry on collecting the data with that additional code, is beyond me.